### DISCUSSION TITLE
Importing  Data from additional sensors

### DISCUSSION THREAD

**Discussion by time4tea** [time4tea/gopro-dashboard-overlay#75] [Ideas]
Following on from #71 - It might be useful to be able to import timeseries data into gopro-overlay so that it can be added to widgets, same as information from a GPX file or GoPro video.

Having had a 2 second think about this - could be terrible idea - it occurred to me that a simple csv file could be used.

We already support access to metrics by name, so accessing arbitrary fields would be relatively straightforward - you'd just have to reference the field in the "metric"

All numeric fields already support time-based interpolation, and these would continue to be interpolated, just as they are now. Interpolation might not be suitable for some fields, but if required we could address that in a different change.

The header row from the CSV file would be used to give fields names, with a required datetime field. Units could be added as per the example.

datetime, field1 (unit),  field2 (unit)
2011-12-01T19:03:17.902,1234.5,2325.0 
...

I think this might be 1) easy to implement 2) compatible with wide number of sensors 

Any thoughts?

**Comment by christian-nils** [comment]
Thanks for continuing the discussion.

Yes, I believe that the CSV file solution is good! 

How would text fields be processed? (e.g.,  overlaying a ML classifier output on the video)

Another thing, how would you handle unsynced data. For instance one sensor output data at 20 Hz, while the other one at 1Hz.

**Comment by time4tea** [comment]
Hadn't really thought of any of these things, hence the discussion. :-)  I do want to keep things as simple as possible.. right now we don't support text at all within the metrics system, so that might not make the first iteration of such a capability...

There are two approaches that I can think of for multiple sensors...1) repeat the sensor data as required across time 2) allow multiple input files. For 1 the creator of the CSV file would just repeat the value as required for each datetime, so a 1Hz value would be just repeated 20 times alongside a 20Hz value.

This might be more difficult for people to create though.

```csv
datetime, 1hz, 20hz
0.00,10,9
0.05,10.2.5
0.10,10,17
...
0.95,10,26
1.00,17,43
....
```

or just two files.
```
datetime,1hz
0.00,10
1.00,17
```

```
datetime,20hz
0.00,10.9
0.05,2.5
....
```

**Reply by christian-nils** [reply]
(Yes, I agree, we should keep it as simple as possible. I was just enumerating the two things I had in mind.)

Regarding the CSV input file, I thought about the second alternative. It seems to be the more elegant way compared with creating duplicated values.

**Comment by PTDreamer** [comment]
Hi, was this ever implement, I'm looking for software to overlay track motorcycle log data including Lean angle and g forces.
